lib/iris/tests/unit/cube_coord_common/test_CFVariableMixin.py

```diff
@@ -0,0 +1,155 @@
+        self.assertIsNone(result)
+
+    def test_fail_colon(self):
+        result = CFVariableMixin.token('nope:')
+        self.assertIsNone(result)
+
+    def test_pass_simple(self):
+        token = 'simple'
+        result = CFVariableMixin.token(token)
+        self.assertEqual(result, token)
+
+    def test_pass_leading_digit(self):
+        token = '123simple'
+        result = CFVariableMixin.token(token)
+        self.assertEqual(result, token)
+
+    def test_pass_mixture(self):
+        token = 'S.imple@one+two_3'
+        result = CFVariableMixin.token(token)
+        self.assertEqual(result, token)
+
+
+class Test_name(tests.IrisTest):
+    def setUp(self):
+        # None token CFVariableMixin
+        self.cf_var = CFVariableMixin()
+        self.cf_var.standard_name = None
+        self.cf_var.long_name = None
+        self.cf_var.var_name = None
+        self.cf_var.attributes = {}
+        self.default = 'unknown'
+        # bad token CFVariableMixin
+        self.cf_bad = CFVariableMixin()
+        self.cf_bad.standard_name = None
+        self.cf_bad.long_name = 'nope nope'
+        self.cf_bad.var_name = None
+        self.cf_bad.attributes = {'STASH': 'nope nope'}
+
+    def test_standard_name(self):
+        token = 'air_temperature'
+        self.cf_var.standard_name = token
+        result = self.cf_var.name()
+        self.assertEqual(result, token)
+
+    def test_long_name(self):
+        token = 'long_name'
+        self.cf_var.long_name = token
+        result = self.cf_var.name()
+        self.assertEqual(result, token)
+
+    def test_var_name(self):
+        token = 'var_name'
+        self.cf_var.var_name = token
+        result = self.cf_var.name()
+        self.assertEqual(result, token)
+
+    def test_stash(self):
+        token = 'stash'
+        self.cf_var.var_name = token
+        result = self.cf_var.name()
+        self.assertEqual(result, token)
+
+    def test_default(self):
+        result = self.cf_var.name()
+        self.assertEqual(result, self.default)
+
+    def test_token_long_name(self):
+        token = 'long_name'
+        self.cf_bad.long_name = token
+        result = self.cf_bad.name(token=True)
+        self.assertEqual(result, token)
+
+    def test_token_var_name(self):
+        token = 'var_name'
+        self.cf_bad.var_name = token
+        result = self.cf_bad.name(token=True)
+        self.assertEqual(result, token)
+
+    def test_token_stash(self):
+        token = 'stash'
+        self.cf_bad.attributes['STASH'] = token
+        result = self.cf_bad.name(token=True)
+        self.assertEqual(result, token)
+
+    def test_token_default(self):
+        result = self.cf_var.name(token=True)
+        self.assertEqual(result, self.default)
+
+    def test_fail_token_default(self):
+        cf_var = CFVariableMixin()
+        cf_var.standard_name = None
+        cf_var.long_name = None
+        cf_var.var_name = None
+        cf_var.attributes = {}
```

Comment:
Is this necessary? I can't see a difference between `cf_var` and `self.cf_var`

Reply:
Yup, I can purge the use of `cf_var`, thanks!